File: Frontend/src/types/audit-log.ts
export interface AuditLogEntry {
  id: string;
  company_id: string;
  user_id: string;
  user_name: string;
  action: string;
  entity_type: string;
  entity_id: string;
  payload_before: string | null;
  payload_after: string | null;
  ip: string | null;
  created_at: string;
}

export interface AuditLogFilters {
  from_date?: string;
  to_date?: string;
  user_id?: string;
  entity_type?: string;
  before_id?: string;
  page_size?: number;
}

export interface AuditLogResponse {
  entries: AuditLogEntry[];
  next_cursor: string | null;
}

export type EntityType = 'sale' | 'user' | 'product' | 'stock_adjustment' | 'expense' | 'cash_closing' | 'payment';

File: Backend/app/schemas/audit_log.py
from datetime import datetime
from typing import Literal, Optional

from pydantic import BaseModel

EntityType = Literal[
    "sale",
    "user",
    "product",
    "stock_adjustment",
    "expense",
    "cash_closing",
    "payment",
]


class AuditLogEntry(BaseModel):
    id: str
    company_id: str
    user_id: str
    user_name: str
    action: str
    entity_type: str
    entity_id: str
    payload_before: Optional[str] = None
    payload_after: Optional[str] = None
    ip: Optional[str] = None
    created_at: datetime

    model_config = {"from_attributes": True}


class AuditLogListResponse(BaseModel):
    entries: list[AuditLogEntry]
    next_cursor: Optional[str] = None
